engine/resource/src/dmsdk/resource/resource.h

```diff
@@ -21,518 +21,501 @@
  *
  * @document
  * @name Resource
- * @namespace dmResource
  * @path engine/resource/src/dmsdk/resource/resource.h
  */
 
-
+#include <stdbool.h>
 #include <stdint.h>
-#include <dmsdk/dlib/align.h>
 #include <dmsdk/dlib/hash.h>
-#include <dmsdk/dlib/hashtable.h>
+#include <dmsdk/dlib/align.h> // DM_ALIGNED
+
+/*#
```

Comment:
The resource.h C header api.

Familiarize yourself with the naming strategies, especially compared to the C++ api.

`struct ResourceFactory;` was previously SResourceFactory, but this approach felt ok. I'm not too worried about name clashes, and we'll notice it on a case by case basis.

The typedef `HResourceFactory` has "Resource" as the prefix.

Similarly, the enums are prefixed. I.e. `enum ResourceResult`.